plugins/ltac/tactic_debug.ml

```diff
@@ -370,7 +372,7 @@ let explain_ltac_call_trace last trace loc =
           strbrk " (with " ++
             prlist_with_sep pr_comma
             (fun (id,c) ->
-              let sigma, env = Pfedit.get_current_context () in
+              let sigma, env = get_current_context () in
```

Comment:
Are the `env` and `sigma` passed as arguments incorrect?

Reply:
Who knows, this is indeed hooked to the reinterpreting code of `ExplainErr` which is in fact already quite messy as it tries to workaround completely broken error handling by the STM; some attempt at fixing a bit the mess is it in #9686 , however OMMV.

We could add an env / sigma parameter to the reinterpretation function but I am not sure it is worth it, we run the risk of overadding sigma / env parameters where like in a few previous PRs.

Reply:
Well, at least we can't keep a function named `get_current_context` which returns the global env, so something has to be done. Not sure what, though.

Reply:
Ok I'll rename it; as to what to here is kind of weird, maybe the traces should keep a copy of the env?

Reply:
@ppedrot any suggestion as to what would be the correct thing to do here? (even if we decide to postpone it)